src/main/java/aws/retrospective/service/CommentService.java

```diff
@@ -0,0 +1,76 @@
+import aws.retrospective.dto.CommentDto;
+import aws.retrospective.entity.Comment;
+import aws.retrospective.repository.CommentRepository;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class CommentService {
+
+    private final CommentRepository commentRepository;
+
+    public CommonApiResponse<List<CommentDto>> getAllComments() {
+        List<Comment> comments = commentRepository.findAll();
+        List<CommentDto> commentDtos = comments.stream()
+            .map(this::convertToDTO)
+            .collect(Collectors.toList());
+        return CommonApiResponse.successResponse(HttpStatus.OK, commentDtos);
+    }
+
+    public CommonApiResponse<CommentDto> getCommentDTOById(Long id) {
+        Comment comment = commentRepository.findById(id)
+            .orElseThrow(() -> new EntityNotFoundException("Comment not found"));
+        CommentDto commentDto = convertToDTO(comment);
+        return CommonApiResponse.successResponse(HttpStatus.OK, commentDto);
+    }
+
+    public CommonApiResponse<Comment> getCommentById(Long id) {
+        Optional<Comment> comment = commentRepository.findById(id);
+        return comment.map(value -> CommonApiResponse.successResponse(HttpStatus.OK, value))
+            .orElseGet(() -> CommonApiResponse.errorResponse(HttpStatus.NOT_FOUND, "Comment not found"));
+    }
+
+    public CommonApiResponse<Comment> createComment(Comment comment) {
+        Comment createdComment = commentRepository.save(comment);
+        return CommonApiResponse.successResponse(HttpStatus.CREATED, createdComment);
+    }
+
+    public CommonApiResponse<Comment> updateComment(Long id, Comment updatedComment) {
+        Comment existingComment = commentRepository.findById(id)
+            .orElseThrow(() -> new EntityNotFoundException("Comment not found"));
+
+        existingComment.getUpdatedDate();
+        Comment savedComment = commentRepository.save(existingComment);
+        return CommonApiResponse.successResponse(HttpStatus.OK, savedComment);
+    }
+
+
+    public CommonApiResponse<Void> deleteComment(Long id) {
+        if (commentRepository.existsById(id)) {
+            commentRepository.deleteById(id);
+            return CommonApiResponse.successResponse(HttpStatus.NO_CONTENT, null);
+        }
+        return CommonApiResponse.errorResponse(HttpStatus.BAD_REQUEST, "Failed to delete comment");
+    }
+
+    private CommentDto convertToDTO(Comment comment) {
+        CommentDto commentDto = new CommentDto();
+        commentDto.setId(comment.getId());
+        commentDto.setComment(comment.getContent());
```

Comment:
```suggestion
        CommentDto commentDto = new CommentDto(comment.getId(), comment.getContent());
```

Setter 대신 생성자를 사용하면 요렇게 개선할 수 있겠네요.
아님 CommentDto의 생성자에서 Comment 객체를 받는 방식도 가능하겠어요.
고민해보세용~!

Reply:
네! 감사합니다!